server.go

```diff
@@ -661,23 +661,43 @@ func (s *Server) Publish(topic string, payload []byte, retain bool, qos byte) er
 	})
 }
 
-// Subscribe adds an inline subscription for the specified topic and handler function.
+// Subscribe adds an inline subscription for the specified client and topic filter with a given handler function.
 // This function allows you to create an internal subscription within the server to handle messages
-// matching the given topic filter. When a message is received on the specified topic,
-// the provided handler function will be called to process the message. returning
-// true if the subscription was new.
-func (s *Server) Subscribe(topic string, handler func(topic string, pk packets.Packet)) bool {
-	return s.Topics.InlineSubscribe(packets.InlineSubscription{
-		Filter:  topic,
-		Handler: handler,
+// matching the given topic filter for a specific client. When a message is received on the specified topic,
+// the provided handler function will be called to process the message. It returns true if the subscription was new.
+func (s *Server) Subscribe(inlineClient, filter string, handler func(client string, pk packets.Packet)) error {
```

Comment:
Because it's only an ID and not an mqtt.Client, I would recommend using `clientID` instead of `inlineClient`.

```suggestion
func (s *Server) Subscribe(clientID, filter string, handler func(client string, pk packets.Packet)) error {
```